examples/fluids/qfunctions/channel.h

```diff
@@ -113,10 +112,14 @@ CEED_QFUNCTION(Channel_Inflow)(void *ctx, CeedInt Q,
   CeedScalar (*v)[CEED_Q_VLA] = (CeedScalar(*)[CEED_Q_VLA])out[0];
   // *INDENT-ON*
   const ChannelContext context = (ChannelContext)ctx;
-  const bool implicit     = context->implicit;
-  const CeedScalar cv     = context->newtonian_ctx.cv;
-  const CeedScalar cp     = context->newtonian_ctx.cp;
-  const CeedScalar gamma  = cp/cv;
+  const bool implicit          = context->implicit;
+  NewtonianIdealGasContext gas = &context->newtonian_ctx;
+  const CeedScalar cv          = gas->cv;
+  const CeedScalar cp          = gas->cp;
+  const CeedScalar gamma       = cp / cv;
+  gas->g[0] = 0.;
+  gas->g[1] = 0.;
+  gas->g[2] = 0.;
```

Comment:
Wouldn't this also set `g` to zero for the other functions too? Making the volumetric forcing term not work for periodic BCs?

Reply:
If the attempt is to remove the potential energy, I think you could just set `x[0] = 0`.

Reply:
I just confirmed that it doesn't affect other QFunctions but the reason I chose to set g to zero is that I thought it was more intuitive and could help with the readability of the code. I was also concerned about  nonzero g[2] and g[2] but I am seeing now that they are set to zero. Anyways, just changed it in [this commit](https://github.com/CEED/libCEED/pull/995/commits/f712d7f79c535eaa1b03e0b87d72a30fbb5c7840) with more comments.

Reply:
> I just confirmed that it doesn't affect other QFunctions

Interesting. Does libCEED make a copy of the struct if you don't set `CeedQFunctionSetContextWritable`? @jedbrown @jeremylt

Reply:
It shouldn't. If you don't set the context read-only, then changes should persist. Could be that this ctx has a copy of the struct in creation

Reply:
I'd thought this was copied out somewhere in the qfunctions, but perhaps not. I don't immediately know why this wouldn't affect other qfunctions.